docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,470 @@
+or ``clang++`` and ``clang`` respectively when using the Open DPC++ Compiler. 
+
+Backends should be enabled by setting ``-DENABLE_<BACKEND_NAME>_BACKEND=True`` for
+each desired backend. By default, only the ``MKLGPU`` and ``MKLCPU`` backends
+are enabled. Multiple backends for multiple device vendors can be enabled at
+once (albeit with limitations when using portBLAS and portFFT). For examples,
+see the section `CMake invocation examples`_. The supported backends for the
+compilers are given in the table at `oneMKL supported configurations table
+<https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#supported-configurations>`_,
+and the CMake option names are given in the table below. Some backends may
+require additional parameters to be set. See the relevant section below for
+additional guidance.
+
+If a backend library supports multiple domains (i.e. BLAS, LAPACK, DFT, RNG), it
+may be desirable to only enable selected domains. For this, the
+``TARGET_DOMAINS`` variable should be set. See the section `TARGET_DOMAINS`_.
+
+By default, the library also additionally builds examples and tests. These can
+be disabled by setting the parameters ``BUILD_FUNCTIONAL_TESTS`` and
+``BUILD_EXAMPLES`` to off. Building the functional tests may require additional
+external libraries. See the section :ref:`building_and_running_tests` for more
+information.
+
+The most important supported build options are:
+
+.. list-table::
+   :header-rows: 1
+
+   * - CMake Option
+     - Supported Values
+     - Default Value 
+   * - ENABLE_MKLCPU_BACKEND
+     - True, False
+     - True      
+   * - ENABLE_MKLGPU_BACKEND
+     - True, False
+     - True      
+   * - ENABLE_CUBLAS_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CUSOLVER_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CUFFT_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_CURAND_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_NETLIB_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCBLAS_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCFFT_BACKEND
+     - True, False
+     - False    
+   * - ENABLE_ROCSOLVER_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_ROCRAND_BACKEND
+     - True, False
+     - False     
+   * - ENABLE_MKLCPU_THREAD_TBB
+     - True, False
+     - True      
+   * - ENABLE_PORTBLAS_BACKEND
+     - True, False
+     - False      
+   * - ENABLE_PORTFFT_BACKEND
+     - True, False
+     - False      
+   * - BUILD_FUNCTIONAL_TESTS
+     - True, False
+     - True      
+   * - BUILD_EXAMPLES
+     - True, False
+     - True      
+   * - TARGET_DOMAINS (list)
+     - blas, lapack, rng, dft, sparse_blas
+     - All domains 
+
+Some additional build options are given in the section `Additional build options`_.
+
+.. _build_target_domains:
+
+TARGET_DOMAINS
+^^^^^^^^^^^^^^
+
+OneMKL supports multiple domains: BLAS, DFT, LAPACK, RNG and sparse BLAS. The
```

Comment:
```suggestion
oneMKL supports multiple domains: BLAS, DFT, LAPACK, RNG and sparse BLAS. The
```

Reply:
Done in [ef02116](https://github.com/oneapi-src/oneMKL/pull/510/commits/ef021162bfa1dd44b8746c8b214cf248b16cbdf9).